### PR TITLE
Joel/nav 7764 convertkit oauth support

### PR DESCRIPTION
**Pull request by joelMege**
### Context
This is the initial set of commits to kick the convertkit-api-ruby gem. It adds support for attaining an authorization token via the OAuth2 flow setup by ConvertKit. It also defines a client that can interact with the ConvertKit API endpoints. At the moment it only supports retrieving the account information. Subsequent work will include support for more of the available API endpoints as listed [here](https://developers.convertkit.com/). 

#### OAuth Class
The `ConvertKit::OAuth` class is used to obtain an access token via the `get_token` method. Per ConvertKit documentation, a client Id, client secret, code, grant type and redirect Uri are required for this call. The client id and client secret are values that can be obtain from ConvertKit. The `code` is obtained by the first phase of the OAuth flow as defined [here](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1).  This class can also refresh a token again per ConvertKit documentation. The `refresh_token` method is very similar to the `get_token` method, but a refresh token is used instead of a `code`.

#### Client Class
The `ConvertKit::Client` class is used to setup a connection based on a authorization token (most likely obtain from the OAuth flow). It is setup to be the main interface to "resources" or the different types of endpoints provided by ConvertKit. The client instantiates a resource, and provides generic `get` and `post` method with some response handling.

#### Specs
I've added support for rspec and include simplecov for test coverage, that includes branch coverage as well. Here's the current coverage report. It's only missing one branch condition in the connection helper. It's a bit tricky to test, but something we can investigate and add in the future.
CleanShot 2023-08-22 at 17 52 24

### CODE REVIEW

File: convertkit-api-ruby.gemspec [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require_relative 'lib/convertkit/version'
+
+Gem::Specification.new do |gem|
+  gem.name        = 'convertkit-api-ruby'
+  gem.version     = ConvertKit::VERSION
+  gem.platform    = Gem::Platform::RUBY
+  gem.authors     = ["Joel Megerssa"]
```
Comment (scribi) [inline convertkit-api-ruby.gemspec]: No to take the credit from youJoel but should this be Mighty Networks ? 

cc: @timherby please let me know if I'm wrong

File: convertkit-api-ruby.gemspec [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  gem.homepage    = 'https://github.com/untitledstartup/convertkit-api-ruby'
+
+  gem.summary     = "Ruby SDK for invoking the ConvertKit API"
```
Comment (scribi) [inline convertkit-api-ruby.gemspec]:
```suggestion
  gem.summary     = 'Ruby SDK for invoking the ConvertKit API'
```

File: lib/convertkit/oauth.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    URL = 'https://app.convertkit.com/'.freeze
+    TOKEN_PATH = 'oauth/token'.freeze
+
+    def initialize(client_id, client_secret, options = {})
+      @id = client_id
+      @secret = client_secret
+      @redirect_uri = options[:redirect_uri]
+      @code = options[:code]
+      @refresh_token = options[:refresh_token]
+      @conn = ConvertKit::ConnectionHelper.get_connection(URL)
```
Comment (scribi) [inline lib/convertkit/oauth.rb]: can we avoid abbreviations?

File: lib/convertkit/version.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module ConvertKit
+  VERSION = "0.0.1"
```
Comment (scribi) [inline lib/convertkit/version.rb]:
```suggestion
  VERSION = '0.0.1'
```

**Review by scribi — APPROVED** [review]
Awesome job @joelMege !, pre-approved with some comments

File: convertkit-api-ruby.gemspec (hunk 1 above)
Comment (joelMege) [reply convertkit-api-ruby.gemspec]: Open to changing this to "Mighty Networks" if that's better.

Comment (timherby) [reply convertkit-api-ruby.gemspec]: I think the gem publisher should be Mighty Networks (though obviously contributions to the repo will be clear by person who contributes to it.

File: lib/convertkit/client.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      handle_response(response)
+    end
+
+    private
+
+    def handle_response(response)
+      if response.success?
+        response.body
+      else
+        raise ConvertKit::APIError, response.body
```
Comment (timherby) [inline lib/convertkit/client.rb]: Should this error include the response status code?  Also, we may eventually want to differentiate error classes making it easier to `rescue` specific ones and retry without needing to inspect them. (for example to retry when rate limited if/when that becomes applicable).

**Review by timherby — APPROVED** [review]
LGTM

File: lib/convertkit/client.rb (hunk 5 above)
Comment (joelMege) [reply lib/convertkit/client.rb]: That's a good point. I'll add the status code to the exception messaging. I'll also add a TODO for differentiating error classes as I can see that being useful!

Comment (timherby) [reply lib/convertkit/client.rb]: BTW, @joelMege here's where I did similar error differentiation for HubSpot: https://github.com/untitledstartup/speckel/commit/3700d18d01171190d554bae7d58dd49815a56cdb#diff-a775350f3eb0724d31d1eb1e24f9bf27fb114b9bbf28cd9caa01489b3ab5e4a5R46

Comment (joelMege) [reply lib/convertkit/client.rb]: Done! https://github.com/untitledstartup/convertkit-api-ruby/pull/1/commits/e1c1e974b88d241e2e30dddb6ea9c9eae7fbebe2

File: lib/convertkit/oauth.rb (hunk 3 above)
Comment (joelMege) [reply lib/convertkit/oauth.rb]: Done! https://github.com/untitledstartup/convertkit-api-ruby/pull/1/commits/6f2faf2ac9c91674155dd6a7390b1f002dfbf2a2

File: convertkit-api-ruby.gemspec (hunk 1 above)
Comment (joelMege) [reply convertkit-api-ruby.gemspec]: Done! https://github.com/untitledstartup/convertkit-api-ruby/pull/1/commits/da0c531ecd8a7bd98959593e3024490bdfa8c32e

File: lib/convertkit/client.rb (hunk 5 above)
Comment (joelMege) [reply lib/convertkit/client.rb]: Oh...cool. Thanks @timherby ! This is a great reference!

**Pull request merged**